geomesa-spark/geomesa-spark-sql/pom.xml

```diff
@@ -61,9 +61,14 @@
         </dependency>
         <dependency>
             <groupId>org.locationtech.geomesa</groupId>
+            <artifactId>geomesa-cqengine_${scala.binary.version}</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.locationtech.geomesa</groupId>
             <artifactId>geomesa-cqengine-datastore_${scala.binary.version}</artifactId>
             <version>${project.version}</version>
-            <scope>test</scope>
+            <scope>compile</scope>
```

Comment:
Hmmm... we might have to try deploying things with and without the new memory-caching options.

It may be that to use the in-memory bits, you have to add some jars to the classpath.  If that's the case, careful documentation is our solution...